wled00/wled.cpp

```diff
@@ -285,6 +285,31 @@ DEBUG_PRINTLN(F("Watchdog: disabled"));
 #endif
 }
 
+#if defined(WLED_ENABLE_BACKGROUND)
+void WLED::backgroundTask(void *parameter)
+{
+  // WLED *instance = static_cast<WLED *>(parameter);
+  
+  for (;;)
+  {
+    // instance->usermods.backgroundLoop();
+    usermods.backgroundLoop();
```

Comment:
To improve, I woud suggest that each `backgroundLoop()` should return the number of millis until it want to be activated again, similar to `return FRAMETIME;` used by effects. This would allow your scheduler to identify "nothing to do" situations, and free the CPU by `vTaskDelay()` if nothing no background loop is due for running.

Reply:
Think this is going to change how I write multithreaded apps for the foreseeable future. Thank you